### ISSUE TITLE
Update to Version 4 - Problems

### ISSUE THREAD

**Issue by fbenji** [will-in-wi/letsencrypt-webfaction#167]
Hi there

I did Update from v3 to v4, but got the following Lines

```
[fenner@web***~]$ letsencrypt_webfaction run
/usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday/options/proxy_options.rb:25: syntax error, unexpected '.' (SyntaxError)
    memoized(:user) { uri&.user && Utils.unescape(uri.user) }
                           ^
/home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday/options/proxy_options.rb:26: syntax error, unexpected '.'
...    memoized(:password) { uri&.password && Utils.unescape(ur...
...                               ^
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday/options.rb:220:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:94:in `block in require_libs'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:93:in `each'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:93:in `require_libs'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:161:in `<module:Faraday>'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:21:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/acme-client-2.0.6/lib/acme/client.rb:3:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/acme-client-2.0.6/lib/acme-client.rb:1:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-4.0.0/lib/letsencrypt_webfaction/certificate_issuer.rb:1:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-4.0.0/lib/letsencrypt_webfaction/application/run.rb:4:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-4.0.0/lib/letsencrypt_webfaction/application.rb:2:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-4.0.0/exe/letsencrypt_webfaction:3:in `<top (required)>'
        from /home/fenner/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `load'
        from /home/fenner/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `<main>'
[fenner@web577 ~]$ letsencrypt_webfaction run
/usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require': /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday/options/proxy_options.rb:25: syntax error, unexpected '.' (SyntaxError)
    memoized(:user) { uri&.user && Utils.unescape(uri.user) }
                           ^
/home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday/options/proxy_options.rb:26: syntax error, unexpected '.'
...    memoized(:password) { uri&.password && Utils.unescape(ur...
...                               ^
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday/options.rb:220:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:94:in `block in require_libs'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:93:in `each'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:93:in `require_libs'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:161:in `<module:Faraday>'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/faraday-1.0.1/lib/faraday.rb:21:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/acme-client-2.0.6/lib/acme/client.rb:3:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/acme-client-2.0.6/lib/acme-client.rb:1:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-4.0.0/lib/letsencrypt_webfaction/certificate_issuer.rb:1:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-4.0.0/lib/letsencrypt_webfaction/application/run.rb:4:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-4.0.0/lib/letsencrypt_webfaction/application.rb:2:in `<top (required)>'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /usr/local/lib/ruby22/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/fenner/.letsencrypt_webfaction/gems/gems/letsencrypt_webfaction-4.0.0/exe/letsencrypt_webfaction:3:in `<top (required)>'
        from /home/fenner/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `load'
        from /home/fenner/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction:23:in `<main>'
[fenner@web***~]$ 
```

**Comment by will-in-wi** [comment]
Hmm. It looks like it grabbed a really new version of Faraday which no longer supports the older version of Ruby.

/me looks for solution.

**Comment by will-in-wi** [comment]
What happens if you run:
```bash
GEM_HOME=$HOME/.letsencrypt_webfaction/gems RUBYLIB=$GEM_HOME/lib gem2.3 install letsencrypt_webfaction
```

And then change
```bash
function letsencrypt_webfaction {
    GEM_HOME=$HOME/.letsencrypt_webfaction/gems PATH=$PATH:$GEM_HOME/bin RUBYLIB=$GEM_HOME/lib ruby2.2 $HOME/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction $*
}
```
to
```bash
function letsencrypt_webfaction {
    GEM_HOME=$HOME/.letsencrypt_webfaction/gems PATH=$PATH:$GEM_HOME/bin RUBYLIB=$GEM_HOME/lib ruby2.3 $HOME/.letsencrypt_webfaction/gems/bin/letsencrypt_webfaction $*
}
```
?

Basically, updating to Ruby 2.3. It looks like all the Webfaction servers support newer Ruby, so it may be time for me to switch.

**Comment by fbenji** [comment]
Hey great, that worked so far. :-))))
Thank you very much for your hard work!!!

William Johnston <notifications@github.com> schrieb am Sa., 18. Apr. 2020,
18:50:

**Comment by will-in-wi** [comment]
My pleasure! Glad it worked for you. I'll update the docs.

**Comment by will-in-wi** [comment]
I just released version 4.0.1 which actually bumps the required Ruby version to 2.4. So if you upgrade, you will want to bump the version again from 2.3. There were some changes from 2.3 to 2.4 which makes it hard to support both, and 2.3 is still pretty old. Now I can support 2.4-2.7.

**Comment by fbenji** [comment]
Great, will look at it. Thanks again for all your help!!! World is a better
place ;-)
Many greetings from Switzerland,
Benjamin

William Johnston <notifications@github.com> schrieb am Sa., 18. Apr. 2020,
19:42:
